src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -474,11 +474,16 @@ internal static partial class Library
 #endregion
 
 #region Unary Operator Implementations
-        private static NumberValue NumericNegate(IRContext irContext, NumberValue[] args)
-        {
+        private static FormulaValue NumericNegate(IRContext irContext, NumberValue[] args)
+        {
+            if (args.Length == 0 || args[0] == null)

```

Comment:
>if (args.Length == 0 || args[0] == null) [](http://example.com/codeflow?start=12&length=40)

How does this happen?

Reply:
it does not. i saw it pass null value in to negate func.

Reply:
That may be the bug ... do you know where those nulls are coming from?

Reply:
This is just masking the symptom of the problem, not addressing the root cause. It should be valid to negate a time/date/datetime value. Something like `-dateValue` should be equivalent to `NumberToDate(Negate(DateToNumber(dateValue)))` - and similar for time/datetime.